### DISCUSSION TITLE
Free / Paid Real Time Data API URL discussion

### DISCUSSION THREAD

**Discussion by rdyvptnm** [aaguiar10/gflows#5] [Q&A] [answered]
Right now I have this data pulled in spread sheet from thinkorswim with excel RTOD function which is free. But excel has its own limitations.

What are the different websites (API_URLs) that can provide either free/paid data?

**Comment by rdyvptnm** [comment]
@aaguiar10 any update on URL link. Let me know if that is some thing you dont want to share. I will try to explore  other options as i am kind of stuck right now,

**Reply by aaguiar10** [reply]
@rdyvptnm,

My API is for internal use only.

**Reply by rdyvptnm** [reply]
@aaguiar10 What are the other options  for me to get real time data?  I tried with the following URL but apparency hash is changing. 
https://www.cboe.com/2c9e0fa3-5570-49b5-938b-d80d4b049293

Appreciate if you can give me any leads even if it is paid.

**Reply by aaguiar10** [reply, marked as answer]
@rdyvptnm,

There are several services (Polygon, Intrinio, Thinkorswim, IBKR, etc.) that show real-time data from their APIs, but require a brokerage account with them and/or paid subscriptions (as well as time to learn their documentation).

I'm working on an update where users can pull option data from a public CBOE API that's currently available ([example w/ SPX](https://cdn.cboe.com/api/global/delayed_quotes/options/_SPX.json)). It'll still be 15 minutes delayed but will speed up the data retrieval process without needing a custom API.

**Comment by rdyvptnm** [comment]
@aaguiar10 that would be great even if it is 15minutes as it is free version.  I have TOS and right now using _"=RTD("tos.rtd", , "OPEN_INT", K4)"_ the RTD function to pull real time data. 

Do you have any example/give some pointers how to pull data from TOS. I found the below API from TOS.
**https://api.tdameritrade.com/v1/marketdata/chains**
will this API fetch in the same format like CBOE csv file format for real time. otherwise, the current parser function might have to change.

Right now TOS is merging with schwab and they have some limitations, so wanted to check if you have tried by any chance.

**Reply by aaguiar10** [reply]
@rdyvptnm,

Since Schwab is merging the TDA API with their own I'm waiting on that release before considering it. I don't know how similar it'll be to TDA's API, but the fetched data should be in json format like most market data APIs.
